### PR TITLE
Add documentation for recovery codes preview feature.

### PR DESCRIPTION
**Pull request by stelewis-redhat**
See code PR at https://github.com/keycloak/keycloak/pull/8730.

### CODE REVIEW

**Review by mposolda — DISMISSED** [review]
@stelewis-redhat Thanks!

@andymunro Do you please also have a chance to review this PR?

File: server_admin/topics/authentication/recovery-codes.adoc
Comment (andymunro) [inline server_admin/topics/authentication/recovery-codes.adoc]: Would this more concise version communicate the same information?

You can configure Recovery codes for two-factor authentication by adding 'Recovery Authentication Code Form' as a two-factor authenticator to your authentication flow. For an example of configuring this authenticator, see the WebAuthn section in xref:configuring-authentication[Configuring authentication].

**Review by andymunro — CHANGES_REQUESTED** [review]
@stelewis-redhat One change suggested.

**Comment by stelewis-redhat** [comment]
> 

Done! @andymunro

**Review by andymunro — APPROVED** [review]
Thanks for the quick fix. thanks, @stelewis-redhat

**Comment by stelewis-redhat** [comment]
> Thanks for the quick fix. thanks, @stelewis-redhat

No problem. Who typically merges, after approval? Is that something I need to do?

EDIT: Oh nevermind, it's not got all the approvals necessary yet.

**Comment by andymunro** [comment]
@stelewis-redhat Actually, I documented the complete process for updating the documentation including approvers.  Yes you need QE and engineering approval before it is merged.  Also, we need to update the RH SSO version after the Keycloak version is merged. Full details are here:  
https://docs.google.com/document/d/1JWiPkLVpyA9TebomUN9wmMM1l1mXO0GhgAkcBLK0xBI/edit#heading=h.k98lwg41g67j

@mposolda, @stianst  The last file changed above is for the Installation Guide. However, starting at Keycloak 17, we have a new installation guide.  https://www.keycloak.org/guides#server Does that guide need to be updated?

**Comment by stelewis-redhat** [comment]
> @stelewis-redhat Actually, I documented the complete process for updating the documentation including approvers. Yes you need QE and engineering approval before it is merged. Also, we need to update the RH SSO version after the Keycloak version is merged. Full details are here: https://docs.google.com/document/d/1JWiPkLVpyA9TebomUN9wmMM1l1mXO0GhgAkcBLK0xBI/edit#heading=h.k98lwg41g67j
> 
> @mposolda, @stianst The last file changed above is for the Installation Guide. However, starting at Keycloak 17, we have a new installation guide. https://www.keycloak.org/guides#server Does that guide need to be updated?

Thanks, @andymunro. Is there a list of people who are QE and engineering? Or is it a group name I can tag?

**Review by mposolda — APPROVED** [review]

**Comment by mposolda** [comment]
@andymunro Thanks for the review! Good point regarding new guide. I see that this page, which is part of new installation guide, will require update https://www.keycloak.org/server/features . However this page is not anymore in the "keycloak/keycloak-documentation" repository, but in the main repository "keycloak/keycloak" inside this file https://github.com/keycloak/keycloak/blob/main/docs/guides/src/main/server/features.adoc . So this PR for "keycloak-documentation" is fine, but the PR for the main codebase will need one small update to that file to reflect the docs.

@stelewis-redhat This PR is fine after Andrew and myself reviewed it. I am approving and I am planning to merge once also codebase PR is going to be merged. Can I request one more small update for the "features.adoc" file in the main repository? I really hope this is the last piece to have this stuff merged! Thanks for all your work on this!

**Comment by stelewis-redhat** [comment]
> @andymunro Thanks for the review! Good point regarding new guide. I see that this page, which is part of new installation guide, will require update https://www.keycloak.org/server/features . However this page is not anymore in the "keycloak/keycloak-documentation" repository, but in the main repository "keycloak/keycloak" inside this file https://github.com/keycloak/keycloak/blob/main/docs/guides/src/main/server/features.adoc . So this PR for "keycloak-documentation" is fine, but the PR for the main codebase will need one small update to that file to reflect the docs.
> 
> @stelewis-redhat This PR is fine after Andrew and myself reviewed it. I am approving and I am planning to merge once also codebase PR is going to be merged. Can I request one more small update for the "features.adoc" file in the main repository? I really hope this is the last piece to have this stuff merged! Thanks for all your work on this!

OK, I'm looking at features.adoc and I don't see where ctx.features.preview is coming from. Gotta figure that out and then I'll push.

EDIT: @mposolda it looks like that features.adoc is driven from Profile.java, so there's no more changes to make?

**Comment by andymunro** [comment]
@stelewis-redhat Let me know if you have an issues updating the RH SSO documentation with the same change using these instructions:   https://docs.google.com/document/d/1JWiPkLVpyA9TebomUN9wmMM1l1mXO0GhgAkcBLK0xBI/edit#heading=h.k98lwg41g67

**Comment by stelewis-redhat** [comment]
> @stelewis-redhat Let me know if you have an issues updating the RH SSO documentation with the same change using these instructions: https://docs.google.com/document/d/1JWiPkLVpyA9TebomUN9wmMM1l1mXO0GhgAkcBLK0xBI/edit#heading=h.k98lwg41g67

@andymunro when it says "Begin the description with a link to the GitHub Issue." which GitHub issue should I use? I didn't create an issue for this project, just the PR.

EDIT: Nevermind. I need to just read the doc.

**Comment by stelewis-redhat** [comment]
@andymunro is there a way to add a label to an issue after the fact? I don't seem to be able to edit the labels in https://github.com/keycloak/keycloak-documentation/issues/1433.

Also, it's unclear which repo I need to modify for RH-SSO. Is it a separate documentation repo, similar to upstream?

**Pull request merged**
